src/ansi-c/literals/convert_float_literal.cpp

```diff
@@ -75,7 +76,7 @@ exprt convert_float_literal(const std::string &src)
     // but these aren't handled anywhere
   }
 
-  ieee_floatt a(to_floatbv_type(result.type()));
+  ieee_floatt a(to_floatbv_type(type));
```

Comment:
How about we go one step further and fix up the return types of the `float_type`, `double_type` etc. to return a `floatbv_typet`? Then this would be ensured at compile-time rather than performing a run-time check?

Reply:
I'd say yes to that, but would make it a separate PR.